frontend/src/components/common/RecentlyViewed/RecentlyViewed.jsx

```diff
@@ -1,21 +1,53 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import Slider from './Slider';
 import styles from './RecentlyViewed.module.scss';
+import { useParams } from 'react-router-dom';
+import { GetProducts } from '../../../services/products/productsAccess.service';
 
-const RecentlyViewed = () => {
-  const recentlyViewed = Array.from(
-    JSON.parse(localStorage.getItem('recentlyViewed') || ''),
+const RecentlyViewed = React.memo(() => {
+  const recentlyViewed = useMemo(
+    () => Array.from(JSON.parse(localStorage.getItem('recentlyViewed') || '')),
+    [],
   );
+  const { myId } = useParams();
+
+  const [receivedData, setReceivedData] = useState([]);
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchDataForAll = async () => {
+      setLoading(true);
+      try {
+        const requests = recentlyViewed.map(async (id) => {
+          return await GetProducts(id);
+        });
+        requests &&
+          (await Promise.all(requests).then((value) => {
+            const result = value.filter((item) => {
+              return item.id !== +myId;
```

Comment:
Maybe it would be better to use filter for `recentlyViewed`. In this case we do not make an unnecessary request to the `GetProducts(id)`
`    const filteredRecentlyViewed = recentlyViewed.filter(
      (item) => item !== myId,
    );`

Reply:
done